atlasdb-config/src/main/java/com/palantir/atlasdb/http/FeignOkHttpClients.java

```diff
@@ -105,14 +105,15 @@ private static Client newOkHttpClient(
             Optional<SSLSocketFactory> sslSocketFactory,
             String userAgent,
             boolean retryOnConnectionFailure) {
-        com.squareup.okhttp.OkHttpClient client = new com.squareup.okhttp.OkHttpClient();
-
-        client.setConnectionSpecs(CONNECTION_SPEC_WITH_CYPHER_SUITES);
-        client.setConnectionPool(new ConnectionPool(CONNECTION_POOL_SIZE, KEEP_ALIVE_TIME_MILLIS));
-        client.setSslSocketFactory(sslSocketFactory.orNull());
-        client.setRetryOnConnectionFailure(retryOnConnectionFailure);
-        client.interceptors().add(new UserAgentAddingInterceptor(userAgent));
-        return new OkHttpClient(client);
+        okhttp3.OkHttpClient.Builder builder = new okhttp3.OkHttpClient.Builder()
+                .connectionSpecs(CONNECTION_SPEC_WITH_CYPHER_SUITES)
+                .connectionPool(new ConnectionPool(CONNECTION_POOL_SIZE, KEEP_ALIVE_TIME_MILLIS, TimeUnit.MILLISECONDS))
+                .retryOnConnectionFailure(retryOnConnectionFailure);
+        if (sslSocketFactory.isPresent()) {
+            builder.sslSocketFactory(sslSocketFactory.get());
```

Comment:
Yeah, `TrustContext` and the appropriate factories are pretty much exactly what I would have written (but wasn't aware of) - thanks for suggesting it!